tools/lsp/preview/element_selection.rs

```diff
@@ -40,6 +52,10 @@ impl ElementRcNode {
             (n.source_file.path().to_owned(), u32::from(n.text_range().start()))
         })
     }
+
+    pub fn is_layout(&self) -> bool {
```

Comment:
Consider this:

```
component MyLayout inherits GridLayout { @children }

component FooBar {
   my-layout := MyLayout {
      Rectangle {
      }
   }
}
```

It is possible that all of that gets inlined together.

In particular, there will be debug node for both `my-layout := MyLayout {`  and for `inherits GridLayout {`, but only the `inherits GridLayout` node will be considered as a layout according to this, i think.
But both are actually layout.

This is very relevant because of HorizontalBox for example.

maybe one fix is in the lower_layout pass to set the layout debug info to both node, so inseat of only using last_mut, iterate over all of them in https://github.com/slint-ui/slint/blob/b4e36f1174e394ca8a8c5bda93a1f1852f300693/internal/compiler/passes/lower_layout.rs#L249 and similar.

Reply:
> So is it safe to copy the debug layout info to all elements on grid_layout_element in the place you pointed out? 

I think it is safe, yes

> This step happens before the inlining, so I assumed at that point there was just one entry in the debug vec. Why are there two entries in the debug vec?

It take place after the first round of inlining